extensions/pipedream/src/services/api.ts

```diff
@@ -0,0 +1,303 @@
+import fetch, { RequestInit, Response } from "node-fetch";
+import { getPreferenceValues } from "@raycast/api";
+import { WorkflowErrorResponse, UserInfo, WorkflowDetails, EventHistory } from "../types";
+import { API_ENDPOINT } from "../utils/constants";
+import { DEMO_ERRORS } from "../utils/demo-data";
+
+/**
+ * Retrieve preferences dynamically so that updates to the API key are reflected
+ * without restarting the extension.
+ */
+function getPreferences() {
+  return getPreferenceValues();
+}
+
+function isDemo(): boolean {
+  return getPreferences().PIPEDREAM_API_KEY === "demo";
+}
```

Comment:
`getPreferences` function is used in only one place so it can be removed.

```suggestion
function isDemo(): boolean {
  return getPreferenceValues<Preferences>().PIPEDREAM_API_KEY === "demo";
}
```